bench/app.py

```diff
@@ -82,47 +85,61 @@ def add_to_excluded_apps_txt(app, bench_path='.'):
 		apps.append(app)
 		return write_excluded_apps_txt(apps, bench_path=bench_path)
 
+
 def write_excluded_apps_txt(apps, bench_path='.'):
 	with open(os.path.join(bench_path, 'sites', 'excluded_apps.txt'), 'w') as f:
 		return f.write('\n'.join(apps))
 
+
 def remove_from_excluded_apps_txt(app, bench_path='.'):
 	apps = get_excluded_apps(bench_path=bench_path)
 	if app in apps:
 		apps.remove(app)
 		return write_excluded_apps_txt(apps, bench_path=bench_path)
 
-def get_app(git_url, branch=None, bench_path='.', skip_assets=False, verbose=False,
-	postprocess = True):
-	# from bench.utils import check_url
-	try:
-		from urlparse import urljoin
-	except ImportError:
-		from urllib.parse import urljoin
 
-	if not check_url(git_url, raise_err = False):
+def get_remote_app(url, branch=None, bench_path='.', skip_assets=False, verbose=False, postprocess=True):
+	if not check_url(url, raise_err=False):
 		orgs = ['frappe', 'erpnext']
 		for org in orgs:
-			url = 'https://api.github.com/repos/{org}/{app}'.format(org = org, app = git_url)
-			res = requests.get(url)
+			api_url = 'https://api.github.com/repos/{org}/{app}'.format(org=org, app=url)
+			res = requests.get(api_url)
 			if res.ok:
-				data    = res.json()
+				data = res.json()
 				if 'name' in data:
-					if git_url == data['name']:
-						git_url = 'https://github.com/{org}/{app}'.format(org = org, app = git_url)
+					if url == data['name']:
+						url = 'https://github.com/{org}/{app}'.format(org=org, app=url)
 						break
 
-	#Gets repo name from URL
-	repo_name = git_url.rsplit('/', 1)[1].rsplit('.', 1)[0]
+	# Gets repo name from URL
+	repo_name = url.rsplit('/', 1)[1].rsplit('.', 1)[0]
 	logger.info('getting app {}'.format(repo_name))
 	shallow_clone = '--depth 1' if check_git_for_shallow_clone() else ''
 	branch = '--branch {branch}'.format(branch=branch) if branch else ''
 
 	exec_cmd("git clone -q {git_url} {branch} {shallow_clone} --origin upstream".format(
-				git_url=git_url,
+				git_url=url,
 				shallow_clone=shallow_clone,
 				branch=branch),
-			cwd=os.path.join(bench_path, 'apps'))
+				cwd=os.path.join(bench_path, 'apps'))
+
+	return repo_name
+
+
+def get_app(url, branch=None, bench_path='.', skip_assets=False, verbose=False, postprocess=True):
+	"""Installs Frappe app from url (git repo or file system) in bench env"""
+
+	try:
+		if os.path.exists(url):
```

Comment:
instead of doing all this, can we add a flag `--local` to handle local files?

Reply:
any specific reason for adding a `local` flag?

Reply:
its easier for the user to just add another flag than to add more complications by checking if the path exists locally before trying to clone from remote.